src/term/transform/flatten.rs

```diff
@@ -177,6 +177,7 @@ fn make_split_rule(old_rule: &Rule, other_rule: &Rule, def_names: &DefNames) ->
           new_pats.push(other_field.clone());
         }
       }
+      (Pattern::Ctr(..), Pattern::Var(None)) => new_pats.push(Pattern::Var(None)),
```

Comment:
We need to also implement `(Pattern::Tup(..), Pattern::Var(None))`, `(Pattern::Num(..), Pattern::Var(None))`, etc, right?

And remove the unreachable on the bottom of the match

Reply:
And add test examples of those other cases